src/Symfony/Component/PasswordHasher/Hasher/UserPasswordHasher.php

```diff
@@ -27,36 +32,78 @@ public function __construct(PasswordHasherFactoryInterface $hasherFactory)
         $this->hasherFactory = $hasherFactory;
     }
 
-    public function hashPassword(UserInterface $user, string $plainPassword): string
+    /**
+     * @param PasswordAuthenticatedUserInterface $user
+     */
+    public function hashPassword($user, string $plainPassword): string
```

Comment:
Is it not possible to change the typehint to PasswordAuthenticatedUserInterface instead of removing it ?

The component PasswordHasher is new and has no releases yet. So the interface can still be changed or am i missing something ?

Reply:
See https://github.com/symfony/symfony/pull/40267#discussion_r579808355

> Using this class requires to follow an upgrade path as well (developers will need to use password_hashers for using it). Accepting both the old type and the new one allow upgrading one step at a time without breakage